### PR TITLE
test(playwright): fix hotkey activation + helper method

### PR DESCRIPTION
**Pull request by skogsmaskin**
### Description

This will fix the Playwright form tests to work on Linux (and CI). There were two issues:

* Platform name was returned incorrectly sometimes (confirmed on local Linux)
* To trigger hotkeys (at least on the PT-input), you have to do that on Linux with a combination of metaDown + hotkeyPress + metaUp (`playwright.page.press('Meta+b')` will not work on Linux). I suspect it's either the way the hotkey event listeners are implemented in the PTE, or it could be a Linux-specific behavior in how keyboard events are handled in the browser.

I have moved the 'focusPTE' method into two separate functions that can target either the PortableTextInput element as a whole (`getFocusedPortableTextInput`), and another that targets the Portable Text Editor within the input (`getFocusedPortableTextEditor`).

With the latter, we can create simpler tests code as most tests before had some lines in each test to do this in order to get to the actual text input.

I have also renamed the `typeInPTEWithDelay` function to `writePortableText` and removed the delay function. The delay can be written more in particular for tests that need to simulate human typing or similar, but it should not be needed in general (I know this helper function has changed a bit through time). Webkit tests are passing now as well, so I have enabled them.

I have disabled the Webkit tests only when platform is `linux` at this time as we don't fully support it.
They are passing on Apple Webkit, so I have enabled them there.

### CODE REVIEW

File: packages/sanity/playwright-ct/tests/utils/testHelpers.tsx [hunk 1]
```diff
@@ -91,5 +90,14 @@ export function testHelpers({
 
       await new Promise((resolve) => setTimeout(resolve, delay || DEFAULT_TYPE_DELAY))
     },
+    /**
+     * Will create a keyboard event of a given meta/hotkey combination
+     * @param key
+     */
+    useHotKey: async (hotkey: string, metaKey: string | undefined = undefined) => {
```
Comment (sjelfull) [inline packages/sanity/playwright-ct/tests/utils/testHelpers.tsx:97]: Nice catch! Can we name this something more descriptive to match the others? The `useX` looks like a React hook.

Like `pressHotkey()`? `toggleHotKey()`?

Comment (skogsmaskin) [reply packages/sanity/playwright-ct/tests/utils/testHelpers.tsx]: Agreed, I have changed it to `toggleHotKey`

**Review by sjelfull — APPROVED** [review]

File: packages/sanity/playwright-ct/tests/utils/testHelpers.tsx [hunk 2]
```diff
@@ -35,29 +35,47 @@ export function testHelpers({
   page: PlaywrightTestArgs['page']
   testInfo: TestInfo
 }) {
-  const platformName = getPlatformName(testInfo.project.name) || process.platform
```
Comment (robinpyon) [inline packages/sanity/playwright-ct/tests/utils/testHelpers.tsx:38]: **suggestion:** given that `getPlatformName` is no longer being used, we can remove it entirely from `testHelpers.tsx`

File: packages/sanity/playwright-ct/tests/formBuilder/PortableTextInput.spec.tsx [hunk 3]
```diff
@@ -36,62 +36,55 @@ test.describe('Activation', () => {
...
 test.describe('Decorators', () => {
   test('Render default styles with keyboard shortcuts', async ({mount, page}, testInfo) => {
-    const {getModifierKey, focusPTE, typeInPTEWithDelay} = testHelpers({
-      page,
-      testInfo,
-    })
+    const {getModifierKey, getFocusedPortableTextEditor, insertPortableText, toggleHotkey} =
+      testHelpers({
+        page,
+        testInfo,
```
Comment (robinpyon) [inline packages/sanity/playwright-ct/tests/formBuilder/PortableTextInput.spec.tsx:42]: **suggestion:** given that `testInfo` is no longer being used in `testHelpers`, consider removing it across the board (which should make these specs a tiny bit slimmer!)

**Review by robinpyon — APPROVED** [review]
Thanks @skogsmaskin! Agree with your take on delayed keypresses

All looks good to me! Only minor point is that we can probably remove the inclusion of `testInfo` across all tests (since that's no longer used in `testHelpers`)

File: packages/sanity/playwright-ct/tests/formBuilder/PortableTextInput.spec.tsx [hunk 4]
```diff
@@ -300,40 +290,44 @@ test.describe('Blocks', () => {
...
     for (const icon of ICONS) {
       await expect(
-        $pteField.getByRole('button').locator(`[data-sanity-icon="${icon}"]`),
+        $portableTextInput.getByRole('button').locator(`[data-sanity-icon="${icon}"]`),
       ).toBeVisible()
     }
 
     // Assertion: ???
-    await expect($pteField.locator('button#block-style-select')).not.toBeVisible()
+    await expect($portableTextInput.locator('button#block-style-select')).not.toBeVisible()
```
Comment (robinpyon) [inline packages/sanity/playwright-ct/tests/formBuilder/PortableTextInput.spec.tsx:306]: **suggestion:** consider breaking this out into a separate test? E.g.

```
  test('Should not display block style button when no block styles are present', async ({
    mount,
    page,
  }, testInfo) => {
    const {getFocusedPortableTextInput} = testHelpers({page, testInfo})
    await mount(<PortableTextInputStory />)
    const $portableTextInput = await getFocusedPortableTextInput('field-bodyStyles')

    // Assertion: Block style button should be hidden
    await expect($portableTextInput.locator('button#block-style-select')).not.toBeVisible()
  })
```

File: packages/sanity/playwright-ct/tests/formBuilder/PortableTextInput.spec.tsx (hunk 3 above)
Comment (skogsmaskin) [reply packages/sanity/playwright-ct/tests/formBuilder/PortableTextInput.spec.tsx:42]: Yeah, I was thinking about that, but didn't want to change too much :)

I think the simpler we can keep this the better.

File: packages/sanity/playwright-ct/tests/formBuilder/PortableTextInput.spec.tsx (hunk 4 above)
Comment (skogsmaskin) [reply packages/sanity/playwright-ct/tests/formBuilder/PortableTextInput.spec.tsx:306]: I have added the test @robinpyon 👍

File: packages/sanity/playwright-ct/tests/formBuilder/PortableTextInput.spec.tsx (hunk 3 above)
Comment (skogsmaskin) [reply packages/sanity/playwright-ct/tests/formBuilder/PortableTextInput.spec.tsx]: Removed that dependency now.

File: packages/sanity/playwright-ct/tests/utils/testHelpers.tsx (hunk 2 above)
Comment (skogsmaskin) [reply packages/sanity/playwright-ct/tests/utils/testHelpers.tsx:38]: Removed it

**Pull request merged**